### DISCUSSION TITLE
Got some advice from a colleague that i should not use register. Should i switch to control instead?

### DISCUSSION THREAD

**Discussion by DeveloperMatheus** [react-hook-form/react-hook-form#10257] [Q&A] [answered]
Context: I made a good amount of reusable components that works perfectly with register (getting the values, validation etc). Those components works directly from the native html tags (input, checkbox).

Later, we had to make a feature that involved those inputs to be dinamically rendered. We achieved this with useFieldArray.

Later, we got some help of a more experienced developer to make a cleaner code and he suggested us to use control instead of register.

**So my question is: Is there a proper way when it comes to creating reusable form components with react-hook-form? Or based in my case, which one is better?** I searched throught the docs, but i didn't have an "orientation" on this matter...

**Comment by leapful** [comment, marked as answer]
@DeveloperMatheus 

You can achieve same readable and clean code base with both `register` and `controller`. If your project is only for web, just use `register` since it's your most comfortable implementation. The `controller` feature might be considered in case that you need to support React Native or similar platform where Web and DOM API is not available to handle interaction event like 'focus', 'blur' ...
